x/wasm/types/types.go

```diff
@@ -309,5 +312,6 @@ func DefaultWasmConfig() WasmConfig {
 		SmartQueryGasLimit: defaultQueryGasLimit,
 		MemoryCacheSize:    defaultMemoryCacheSize,
 		ContractDebugMode:  defaultContractDebugMode,
+		SmartQueryDepth:    defaultSmartQueryDepth,
```

Comment:
This can be overridden by genesis? Or per machine?

```
SmartQueryGasLimit: defaultQueryGasLimit,
MemoryCacheSize:    defaultMemoryCacheSize,
ContractDebugMode:  defaultContractDebugMode,
```

should not be consensus breaking, in that different nodes may have different values and they will handle external queries differently, have different performance profiles, etc but will return the same result for all transactions.

`SmartQueryDepth` will actually abort transactions and could cause consensus issues if different on different validator nodes.

Either we only allow setting it in the Go binary, or we set it in genesis (as x/params?). WasmConfig was read from a local toml file, right?